avocado/plugins/runner_nrunner.py

```diff
@@ -50,11 +51,16 @@ def initialize(self):
                                  help_msg=help_msg,
                                  key_type=bool)
 
+        status_server_default = '127.0.0.1:8888'
+        address, port = status_server_default.split(':')
+        if not network.is_port_free(int(port), address):
+            status_server_default = "%s:%s" % (address,
+                                               network.find_free_port())
         help_msg = ('URI for listing the status server. Usually '
                     'a "HOST:PORT" string')
         settings.register_option(section=section,
                                  key='status_server_listen',
-                                 default='127.0.0.1:8888',
+                                 default=status_server_default,
```

Comment:
I find it strange to have a "random port" as the default value. What about letting the user choose the port when creating a job inside a test? I'm not sure here.